src/main/java/dev/openfeature/sdk/MutableTrackingEventDetails.java

```diff
@@ -0,0 +1,45 @@
+package dev.openfeature.sdk;
+
+import dev.openfeature.sdk.internal.ExcludeFromGeneratedCoverageReport;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.experimental.Delegate;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * MutableTrackingEventDetails represents data pertinent to a particular tracking event.
+ */
+@EqualsAndHashCode
+@ToString
+public class MutableTrackingEventDetails implements TrackingEventDetails {
```

Comment:
I think to be consistent with the rest if the SDK we should also implement a `ImutableTrackingEventDetails`.

Not exactly a blocker for me but I think it would be a good idea.